### PR TITLE
[rtloader] python instance memory tracking

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR adds support for reporting python check instance memory usage.

### Motivation

Adding tooling for additional insights into the python runtime, and memory tracking in general.

### Additional Notes

This PR was based off #3961 - so merge that first, then change the `base` to `master`, then this.

Unsure if we should set the `ignored`: `pympler.asizeof.asizeof(*objs, **opts)` option to `True`, it might make sense. See here: https://pythonhosted.org/Pympler/asizeof.html

### CODE REVIEW

File: cmd/agent/api/agent/agent.go [hunk 1]
```diff
@@ -219,6 +220,25 @@ func getFormattedStatus(w http.ResponseWriter, r *http.Request) {
...
+func getCheckMemoryStatus(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	component := vars["component"]
+	switch component {
+	case "py":
+		jsonStatus, err := status.GetPythonCheckMemoryUsage(common.Coll)
+		if err != nil {
+			log.Errorf("unable to collect python check memory usage: %s", err.Error())
+			http.Error(w, err.Error(), 500)
+		}
```
Comment (ogaca-dd) [inline cmd/agent/api/agent/agent.go]: Do we need an `else` here?

File: pkg/status/status_python.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		if err != nil {
+			log.Errorf("Error collecting size for instance %v: %v", pyCheck.ID(), err)
+			continue
+		}
+
+		_, ok = stats[pyCheck.String()]
+		if !ok {
+			stats[pyCheck.String()] = make(map[check.ID]int64)
+		}
+		stats[pyCheck.String()][pyCheck.ID()] = size
```
Comment (ogaca-dd) [inline pkg/status/status_python.go]: Is the composite key `pyCheck.String()` + `pyCheck.ID()` unique in this function? Is it possible to have `stats["foo"][42]` twice for example?

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/status/status_python.go (hunk 2 above)
Comment (truthbk) [reply pkg/status/status_python.go]: No, that shouldn't be possible as the check ID should always be unique to an instance.

File: cmd/agent/api/agent/agent.go (hunk 1 above)
Comment (truthbk) [reply cmd/agent/api/agent/agent.go]: No, but what I am missing is a return after writing the `500`. Let me clean that up! Thanks for the catch.

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**